app/models/notification.rb

```diff
@@ -40,6 +42,12 @@ def to_h
     }
   end
 
+  def backup_subject_name
+    return unless subject
+    backup_name = (subject.try(:name) || subject.try(:description))
+    self.options[:subject] = backup_name if backup_name
```

Comment:
Does `options[:subject]` get used somewhere?

Reply:
yeah, it's quite non-obvious.

The UI calls `#to_h`.  This calls `text_bindings` which calls `text_bindings_dynamic` which walks options key/values and will be used when the `subject` (vm or thing the notification is based on) is no longer existing.  See line 90 👇 

I was debating rewriting this to make this more clear but the changes were too complicated for this bug fix.